overlord/snapstate/check_snap.go

```diff
@@ -99,40 +99,95 @@ func checkVersion(version string) bool {
 	return true
 }
 
-var openSnapFile = backend.OpenSnapFile
+type ErrSnapNeedsMode struct {
+	Mode snap.ConfinementType
+	Snap string
+}
 
-// checkSnap ensures that the snap can be installed.
-func checkSnap(st *state.State, snapFilePath string, si *snap.SideInfo, curInfo *snap.Info, flags Flags) error {
-	// This assumes that the snap was already verified or --dangerous was used.
+func (e *ErrSnapNeedsMode) Error() string {
+	return fmt.Sprintf("snap %q requires %s or confinement override", e.Snap, e.Mode)
+}
 
-	s, _, err := openSnapFile(snapFilePath, si)
-	if err != nil {
-		return err
-	}
+type ErrSnapNeedsClassicSystem string
```

Comment:
this seems a bit too cute, I would have used (given that is exported) a struct with Snap anyway